ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -92,6 +92,13 @@
   private volatile boolean responseComplete = false;
   // marked as true if force close is required because close() was called.
   private volatile boolean forceClose = false;
+  // the ResponseStatus that will be sent (or has been sent) as a part of the response.
+  private volatile ResponseStatus responseStatus = ResponseStatus.Ok;
```

Comment:
Yeah. If no body is written but `onResponseComplete()` is called directly without an `exception`, it is assumed to be an OK response (can't be any other status). 

The interface definition also says so.